src/rest/server_service/path_predicate.rs

```diff
@@ -0,0 +1,79 @@
+use actix_web::pred::Predicate;
+use actix_web::Request;
+use regex::{self, Regex};
+
+pub struct PathPredicate {
+    regex: Regex,
+}
+
+impl PathPredicate {
+    // The pattern should be a full URI, with or without slash on the beginning.
+    // Wildcards in form of `{some_name}` are allowed, they accept everything except `/`.
+    // The predicate will accept only requests with URIs fully matching the pattern.
+    pub fn for_pattern(pattern: &str) -> Self {
```

Comment:
Seems legit now. My last question about this is: Doesn't Actix have built-in path matchers for this?

Reply:
Yes and no. It has excellent path matcher for routes inside `App`. Unfortunately we need to have multiple `App`s and we must decide which one of them should receive the request and it's an irreversible routing decision. We must have multiple `App`s, because each of them captures different context and we also want to keep them composable and `App`s are good building blocks.

Reply:
Maybe each of the apps could have a simple predicate that matches just the prefix as a static string and we can do away with regex completely?

Reply:
It worked that way before introduction of `block/{block_id}` and `block/{block_id}/next_id`.